table/block_based/filter_policy.cc

```diff
@@ -393,14 +388,12 @@ class FastLocalBloomBitsBuilder : public XXPH3FilterBitsBuilder {
     TEST_SYNC_POINT_CALLBACK("XXPH3FilterBitsBuilder::Finish::TamperFilter",
                              &TEST_arg_pair);
 
-    Slice rv(mutable_buf.get(), len_with_metadata);
-    *buf = std::move(mutable_buf);
+    assert(mutable_buf.get_deleter().allocator == allocator);
```

Comment:
Not quite sure about the intention of this assertion - is there any wrong behavior this assertion is for?